src/util/ped_editor.hpp

```diff
@@ -1,170 +1,287 @@
-		PED_COMP_HAND, 
-		PED_COMP_FEET, 
-		PED_COMP_TEEF, 
-		PED_COMP_ACCS, 
-		PED_COMP_TASK, 
-		PED_COMP_DECL, 
-		PED_COMP_JBIB 
-	};
-	inline constexpr int prop_ids[5] = {
-		PED_PROP_HATS,
-		PED_PROP_GLASSES,
-		PED_PROP_EARS,
-		PED_PROP_WATCHES,
-		PED_PROP_BRACELETS
-	};
-
-	inline int component_drawable_ids[12] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
-	inline int component_texture_ids[12] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
-	inline int prop_drawable_ids[5] = { 0, 0, 0, 0, 0 };
-	inline int prop_texture_ids[5] = { 0, 0, 0, 0, 0 };
-
-	inline bool get_ped_components()
-	{
-		Ped ped = self::ped;
-
-		for (int i = 0; i < countof(component_ids); ++i)
-		{
-			g_fiber_pool->queue_job([ped, i]
-				{
-					component_drawable_ids[i] = PED::GET_PED_DRAWABLE_VARIATION(ped, component_ids[i]);
-					component_texture_ids[i] = PED::GET_PED_TEXTURE_VARIATION(ped, component_ids[i]);
-				});
-		}
-
-		return true;
-	}
-
-	inline bool set_ped_components()
-	{
-		Ped ped = self::ped;
-
-		for (int i = 0; i < countof(component_ids); ++i)
-		{
-			g_fiber_pool->queue_job([ped, i]
-				{
-					PED::SET_PED_COMPONENT_VARIATION(ped, i, component_drawable_ids[i], component_texture_ids[i], 2);
-				});
-		}
-
-		return true;
-	}
-	
-	inline bool get_ped_props()
-	{
-		Ped ped = self::ped;
-
-		for (int i = 0; i < countof(prop_ids); ++i)
-		{
-			g_fiber_pool->queue_job([ped, i]
-				{
-					prop_drawable_ids[i] = PED::GET_PED_PROP_INDEX(ped, prop_ids[i]);
-					prop_texture_ids[i] = PED::GET_PED_PROP_TEXTURE_INDEX(ped, prop_ids[i]);
-				});
-		}
-
-		return true;
-	}
-
-	inline bool set_ped_props()
-	{
-		Ped ped = self::ped;
-		Hash model = ENTITY::GET_ENTITY_MODEL(ped);
-
-		if (model != MISC::GET_HASH_KEY("mp_m_freemode_01") &&
-			model != MISC::GET_HASH_KEY("mp_f_freemode_01"))
-			return false;
-
-		for (int i = 0; i < countof(prop_ids); ++i)
-		{
-			if (prop_drawable_ids[i] == -1)
-			{
-				g_fiber_pool->queue_job([ped, i]
-					{
-						PED::CLEAR_PED_PROP(ped, i);
-					});
-			}
-			else
-			{
-				g_fiber_pool->queue_job([ped, i] {
-					PED::SET_PED_PROP_INDEX(ped, prop_ids[i], prop_drawable_ids[i], prop_texture_ids[i], TRUE);
-					});
-			}
-		}
-
-		return true;
-	}
-
-	inline const char* get_ped_component_name(int component_id)
-	{
-		const char* component_name = "OOF";
-
-		g_fiber_pool->queue_job([component_id]
-			{
-				Hash component_hash = FILES::GET_HASH_NAME_FOR_COMPONENT(self::id, component_id, component_drawable_ids[component_id], component_texture_ids[component_id]);
-				PedComponent ped_component;
-				const char* component_name{};
-
-				FILES::GET_SHOP_PED_COMPONENT(component_hash, (Any*)&ped_component);
-				component_name = HUD::GET_FILENAME_FOR_AUDIO_CONVERSATION(ped_component.m_textLabel);
-			});
-		
-		return component_name;
-	}
-}
+#pragma once
+#include "natives.hpp"
+#include "pointers.hpp"
+#include "script.hpp"
+
+template <typename t, size_t szof>
+constexpr int countof(t (&array)[szof]) { return szof; }
```

Comment:
just use ImGui's `IM_ARRAYSIZE`

Reply:
Just a simple `sizeof()` should work there as well, the compiler should be smart enough to substitute that to the correct size.